### ISSUE TITLE
Spatial component should use floats for distance

### ISSUE THREAD

**Issue by wu-edward** [solariumphp/solarium#1095]
**Solarium version(s) affected**: 6.3.1
**Solr version**: 8.11.2
**Solr mode**: standalone

**Description**
The spatial component only allows integer values for distance. This can cause a lack of precision if there is a need to convert between units (such as mi -> km). See https://www.drupal.org/project/search_api_solr/issues/3389956 for more details.

**How to reproduce**

**Possible Solution**
Change \Solarium\Component\Spatial getDistance, and setDistance methods to set and return distance as float

**Additional context**

**Comment by thomascorthals** [comment]
Good catch. The parameter is indeed parsed as a `double` in Solr.

https://github.com/apache/solr/blob/4c80cc2ef9e50915fb7aaa6e9cfe042a27d2c44d/solr/core/src/java/org/apache/solr/search/SpatialFilterQParser.java#L59-L61

```java
    double dist = -1;
    String distS = getParam(SpatialParams.DISTANCE);
    if (distS != null) dist = Double.parseDouble(distS);
```

**Comment by thomascorthals** [comment]
Fixed in #1096.
